pkgs/development/python-modules/knocki/default.nix

```diff
@@ -0,0 +1,60 @@
+{
+  aiohttp,
+  aioresponses,
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  mashumaro,
+  orjson,
+  poetry-core,
+  pythonOlder,
+  pytestCheckHook,
+  pytest-aiohttp,
+  syrupy,
+  yarl,
+}:
+
+buildPythonPackage rec {
+  pname = "knocki";
+  version = "0.3.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.11";
+
+  src = fetchFromGitHub {
+    owner = "swan-solutions";
+    repo = "knocki-homeassistant";
+    rev = "v${version}";
+    hash = "sha256-tWtANsujAcdIddoUBrKWIPfiPzDqWhW94Goz0QQ2BfE=";
+  };
+
+  nativeBuildInputs = [ poetry-core ];
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "--cov" ""
+  '';
```

Comment:
wouldn't it be better to provide `pytest-cov` as a `nativeCheckInputs`?

Reply:
I can certainly do that!

Only issue being the project uses the plugin `covdefaults` which isn't yet in nixpkgs

```
[tool.coverage.run]
plugins = ["covdefaults"]
source = ["knocki"]
```

I can instead do this:

```
postPatch = ''
    substituteInPlace pyproject.toml \
      --replace-fail "[\"covdefaults\"]" "[]"
  '';
```

To remove the plugin usage until `covdefaults` is in `nixpkgs`

Would that be better?

Reply:
Went with this approach - let me know your thoughts.

Reply:
We do not need to measure coverage because we test to make sure it works.
Just remove `addopts = "--cov"`, please.

Reply:
Removed `adopts = "--cov"` in the latest commit. ✅